packages/headless/src/features/insight-user-actions/insight-user-actions-preprocessing.test.ts

```diff
@@ -0,0 +1,962 @@
+          expect(timeline.session?.actions[1].actionType).toEqual(
+            UserActionType.TICKET_CREATION
+          );
+        });
+      });
+    });
+
+    describe('when it does not find a case creation session', () => {
+      describe('when the ticket creation date comes before all the sessions', () => {
+        it('should return the timeline with the ticket creation date set as the current session timestamp and following sessions', async () => {
+          const mockRawActions = [...fakeActions];
+          const expectedTimeline = {
+            precedingSessions: [],
+            session: {
+              start: new Date('Mon Mar 28 2022 16:10:00 GMT').valueOf(),
+              end: new Date('Mon Mar 28 2022 16:10:00 GMT').valueOf(),
+              actions: [
+                {
+                  actionType: 'TICKET_CREATION',
+                  timestamp: new Date('Mon Mar 28 2022 16:10:00 GMT').valueOf(),
+                  eventData: {},
+                },
+              ],
+            },
+            followingSessions: [
+              {
+                start: new Date('Tue Mar 29 2022 9:24:30 GMT').valueOf(),
+                end: new Date('Tue Mar 29 2022 9:24:30 GMT').valueOf(),
+                actions: [
+                  {
+                    actionType: 'CUSTOM',
+                    timestamp: new Date(
+                      'Tue Mar 29 2022 9:24:30 GMT'
+                    ).valueOf(),
+                    eventData: {
+                      type: 'smartSnippetSuggestions',
+                      value: 'likeSmartSnippet',
+                    },
+                    searchHub: 'in-product-help',
+                    document: {},
+                  },
+                ],
+              },
+              {
+                start: new Date('Tue Mar 29 2022 8:51:00 GMT').valueOf(),
+                end: new Date('Tue Mar 29 2022 8:51:00 GMT').valueOf(),
+                actions: [
+                  {
+                    actionType: 'CLICK',
+                    timestamp: new Date(
+                      'Tue Mar 29 2022 8:51:00 GMT'
+                    ).valueOf(),
+                    eventData: {},
+                    searchHub: 'in-product-help',
+                    document: {
+                      title: 'title',
+                      uriHash: 'someHash',
+                    },
+                  },
+                ],
+              },
+            ],
+          };
+          const mappedAndSortedActions =
+            mapAndSortActionsByMostRecent(mockRawActions);
+          // Date far back before the first session
+          const ticketCreationDate = createRelativeDate(
+            firstSessionDate,
+            -1000,
+            0
+          ).getTime();
+
+          const timeline = splitActionsIntoTimelineSessions(
+            mappedAndSortedActions,
+            ticketCreationDate,
+            actionsToExclude
+          );
+
+          expect(timeline).toEqual(expectedTimeline);
```

Comment:
same deep equal comment

Reply:
As mehdi said in the other comment, the `toEqual` is the deep equality in jest.
`toBe` is the shallow one

<img width="1067" alt="image" src="https://github.com/user-attachments/assets/9386c529-ca0c-467a-9a00-994838851d61">

Reply:
🤦 I was thinking we were in Cypress xD

You're right! nvm me!